### PR TITLE
Add keycloak package.

### PR DESCRIPTION
**Pull request by ainmosni**
This PR adds the keycloak package, which contains the following:

* More genericised functions for the invitations backend.
* User query functions.
* PII query functions.

### CODE REVIEW

File: services/invitations/pkg/backends/keycloak/backend.go [hunk 1]
```diff
@@ -3,31 +3,28 @@ package keycloak
...
+	"github.com/owncloud/ocis/v2/ocis-pkg/keycloak"
 	"github.com/owncloud/ocis/v2/ocis-pkg/log"
 	"github.com/owncloud/ocis/v2/services/invitations/pkg/invitations"
 )
 
 const (
-	idAttr       = "OWNCLOUD_ID"
-	userTypeAttr = "OWNCLOUD_USER_TYPE"
-	userTypeVal  = "Guest"
+	userType = "Guest"
```
[the remark this answers is not in the document]
Comment (ainmosni) [reply services/invitations/pkg/backends/keycloak/backend.go:15]: No, they just moved to another place. :)

**Review by kobergj — APPROVED** [review]
Just minors. Rest looks great :+1:

File: ocis-pkg/keycloak/client.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+	"context"
+	"crypto/tls"
+	"fmt"
+
+	"github.com/Nerzal/gocloak/v13"
+	libregraph "github.com/owncloud/libre-graph-api-go"
+)
+
+// Some attribute constants.
+// TODO: Make these configurable in the future.
```
[the remark this answers is not in the document]
Comment (ainmosni) [reply ocis-pkg/keycloak/client.go:14]: 'm torn on this, on one hand, I think we have too many configurables as it is, but I can also make a case for making this configurable. So I took the cowards way out, and moved this decision to the future. :)

File: ocis-pkg/keycloak/client.go [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+const (
+	idAttr       = "OWNCLOUD_ID"
+	userTypeAttr = "OWNCLOUD_USER_TYPE"
```
[the remark this answers is not in the document]
Comment (ainmosni) [reply ocis-pkg/keycloak/client.go]: Done

File: ocis-pkg/keycloak/client.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kobergj) [reply ocis-pkg/keycloak/client.go:14]: Alright. More fun for the future-us

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>Core-API-Tests-ocis-storage-4</strong>](https://drone.owncloud.com/owncloud/ocis/20963/59/1) failed. Further test are cancelled...

**Pull request merged**